api/python/quilt3/packages.py

```diff
@@ -619,6 +619,10 @@ def download_manifest(dst):
 
             pkg = cls._from_path(local_pkg_manifest)
             pkg._origin = PackageRevInfo(str(registry.base), name, top_hash)
+
+            if pkg.top_hash != top_hash:
```

Comment:
that might be OK, but are you sure we should put manifest to the cache if this check fails?

Reply:
It's safe to do that right now, since it checks the hash regardless of where the package is being loaded from - local, S3, or cache.

I could change it to only check hash when loading from S3, and before putting it into the cache. Then we'll trust any local package (not sure we should?), and anything in the cache (probably fine, except for packages that were already cached before this PR).